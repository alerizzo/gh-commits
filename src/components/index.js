export { default as Skeleton } from './Skeleton/Skeleton';

export { default as TextInput } from './TextInput/TextInput';
export { default as Button } from './Button/Button';

export { default as Layout } from './Layout/Layout';

export { default as CommitsList } from './CommitsList/CommitsList';
export { default as RepositorySearch } from './RepositorySearch/RepositorySearch';
