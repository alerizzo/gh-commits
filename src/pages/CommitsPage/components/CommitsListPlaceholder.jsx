import React from 'react';
import { CommitsList } from 'components';

function CommitsListPlaceholder({ message, loading }) {
  return (
    <div className="CommitsListPlaceholder">
      <CommitsList.Skeleton rows={3} loading={loading} />
      {message && <div className="overlay is-size-5 has-text-weight-bold">{message}</div>}
    </div>
  );
}

export default CommitsListPlaceholder;
