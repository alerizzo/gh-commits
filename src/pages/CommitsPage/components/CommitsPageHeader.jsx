import React from 'react';
import { TextInput } from 'components';
import CommitRow from 'components/CommitsList/CommitRow';

export default function CommitsPageHeader({ onSearch, repository, disabled, showTableHeader }) {
  return (
    <div className="CommitsPageHeader">
      <h1 className="title is-3">Commits list</h1>
      <div className="field">
        <label className="label is-size-6">
          Search
          {repository && (
            <span className="has-text-weight-light">
              {' in '}
              {repository.owner}'s {repository.repository} ({repository.refName})
            </span>
          )}
        </label>
        <TextInput style={{ maxWidth: '50%' }} disabled={disabled} placeholder="Search ..." />
      </div>
      {showTableHeader && (
        <div className="CommitsPageHeader_TableHeader">
          <CommitRow.Header />
        </div>
      )}
    </div>
  );
}
