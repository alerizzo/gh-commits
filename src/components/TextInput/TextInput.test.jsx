import React from 'react';
import renderer from 'react-test-renderer';

import { TextInput } from 'components';

describe('<TextInput />', () => {
  const defaultProps = {};
  const wrapper = renderer.create(<TextInput {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<TextInput isInvalid invalidMessage="Error text" />', () => {
  const defaultProps = {
    isInvalid: true,
    invalidMessage: 'Error text',
  };
  const wrapper = renderer.create(<TextInput {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<TextInput isLoading />', () => {
  const defaultProps = {
    isLoading: true,
  };
  const wrapper = renderer.create(<TextInput {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
