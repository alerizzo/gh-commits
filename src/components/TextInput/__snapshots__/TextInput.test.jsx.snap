// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> render 1`] = `
<div
  className="TextInput control "
  data-error={null}
>
  <input
    className="input  "
    type="text"
  />
</div>
`;

exports[`<TextInput isInvalid invalidMessage="Error text" /> render 1`] = `
<div
  className="TextInput control "
  data-error="Error text"
>
  <input
    className="input is-danger "
    type="text"
  />
</div>
`;

exports[`<TextInput isLoading /> render 1`] = `
<div
  className="TextInput control is-loading"
  data-error={null}
>
  <input
    className="input  "
    type="text"
  />
</div>
`;
